### PR TITLE
include/rdma: New options to fi_getopts for EFA

### PR DESCRIPTION
**Pull request by lrbison**
* Added three new options for fi_getopts in fi_ext.h: FI_OPT_EFA_EMULATED_READ FI_OPT_EFA_EMULATED_WRITE FI_OPT_EFA_EMULATED_ATOMICS

 * Added documentation in fi_efa.7.md These options can be used by libfabric users to query if an EFA endpoint is using emulation or has hardware support for the Read, Write, and Atomics operations.

### CODE REVIEW

**Comment by j-xiong** [comment]
Provider specific options have their own range and should not be here. See `fi_ext.h` for examples of how provider specific options are defined.

**Review by shefty — CHANGES_REQUESTED** [review]
Provider defines go into fi_ext.h.

**Comment by lrbison** [comment]
Thank you @shefty and @j-xiong, I'm sorry for not realizing that.

I've moved the documentation to fi_efa's .md file and the definitions to fi_ext.h, and updated the commit message.

File: man/fi_efa.7.md [hunk 1]
```diff
@@ -97,6 +97,11 @@ provider for AWS Neuron or Habana SynapseAI.
   the endpoint is enabled. Otherwise, the call will fail. Also note that this
   option only applies to RDM endpoint.
 
+*FI_OPT_EFA_EMULATED_READ, FI_OPT_EFA_EMULATED_WRITE, FI_OPT_EFA_EMULATED_ATOMICS - bool*
+: These options only apply to the fi_getopt() call.
+  They are used to query the EFA provider to determine if the endpoint is  emulating Read, Write, and Atomic operations (return value is true), or if
```
Comment (wenduwan) [inline man/fi_efa.7.md:102]: Telling from the rest of the file, this line has probably exceeded the max length.

Comment (lrbison) [reply man/fi_efa.7.md]: Looks like a missed line ending and didn't notice because I keep my window narrow.  Thank you, fixed!

**Comment by lrbison** [comment]
@shefty gentle ping on this fi_getopts change, is the new location in fi_ext.h appropriate?

**Review by shefty — APPROVED** [review]

**Comment by shefty** [comment]
Yes, this is good -- thanks!

**Review by wckzhang — APPROVED** [review]

**Pull request merged**